47.md

```diff
@@ -63,6 +63,267 @@ Response:
+{
+    "method": "pay_keysend",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "pubkey": "03...", // payee pubkey, required
+        "message": "some message", // keysend message, optional
+        "preimage": "0123456789abcdef...", // preimage of the payment, optional
+        "tlv_records: [ // tlv records, optional
+            {
+                "type": 5482373484, // tlv type
+                "value": "fajsn341414fq" // hex encoded tlv value
+            }
+        ]
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_keysend",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+        "payment_hash": "31afdf1..", // payment hash of the payment
+    }
+}
+```
+
+### Make Invoice
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### Lookup Invoice
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "lookup_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "invoice": "lnbc50n1...", // the encoded invoice
+        "paid": true // true if the invoice has been paid
+    }
+}
+```
+
+
+### List Invoices
+
+Request:
+```jsonc
+{
+    "method": "list_invoices",
+    "params": {
+        "from": 1693876973, // starting timestamp in seconds since epoch, optional
+        "until": 1693876973, // ending timestamp in seconds since epoch, optional
+        "limit": 10, // maximum number of invoices to return, optional
+        "offset": 0, // offset of the first invoice to return, optional
+        "unpaid": true // include unpaid invoices, optional, default false
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "list_invoices",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": [
```

Comment:
Returning an array directly means we cannot easily extend the response in the future. Would it make sense to return an object e.g.

```json
result: {
  invoices: [],
  // ... other properties here
}
```

LND also does this: https://api.lightning.community/api/lnd/lightning/list-invoices/index.html

Reply:
did the same for list_payments too